### PR TITLE
Ref currency

### PR DESCRIPTION
**Pull request by Overtorment**
in this PR we:

* get rid of timers, as we suspect they cause issues, particularly tor freezes (maybe jsc on android doesn't like timers, especially frequent ones like we had, 12sec). exchange rate is now fetched upon app startup, and when app transitions from background to foreground. as before, its not updating rate if its called sooner than 30 min (that hasn't changed).
* simplify & refactor currency module
* throw away frisbee in favor of bare fetch for currency exchange API

cc @ncoelho 

needs proper testing 

PS. i would like to pat myself on the back yet again for having tests, they help a lot

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -262,43 +263,43 @@ const App = () => {
...
-            setClipboardContentType(ClipboardContentType.BITCOIN);
-          } else if (isLightningInvoice || isLNURL) {
-            setClipboardContentType(ClipboardContentType.LIGHTNING);
-          } else if (isBothBitcoinAndLightning) {
-            setClipboardContentType(ClipboardContentType.BITCOIN);
-          }
-          setIsClipboardContentModalVisible(true);
+    if (wallets.length === 0) return;
+    if ((appState.current.match(/background/) && nextAppState === 'active') || nextAppState === undefined) {
+      setTimeout(() => A(A.ENUM.APP_UNSUSPENDED), 2000);
```
Comment (marcosrdz) [inline App.js:268]: I can't remember why this is needed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/q5dq7bqhwg28bkpu8n1jhjzc04?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8218)

File: App.js (hunk 1 above)
Comment (Overtorment) [reply App.js:268]: what exactly? `A`? its analytics event

Comment (marcosrdz) [reply App.js:268]: Ah my bad i misread it. Nvm

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xtf5tw6dcg5vnd6ddvqbqkdtrm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8219)

**Review by marcosrdz — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**